### PR TITLE
Wasm cross compiler

### PR DESCRIPTION
**Pull request by directhex**
Leverage the infra built for the Android cross compiler, to make a Wasm cross compiler.

### CODE REVIEW

**Comment by directhex** [comment]
Official build at https://dev.azure.com/dnceng/internal/_build/results?buildId=956406&view=results

**Review by steveisok — APPROVED** [review]

File: eng/Subsets.props [hunk 1]
```diff
@@ -45,7 +45,7 @@
     <DefaultMonoSubsets Condition="'$(MonoEnableLLVM)' == 'true' and '$(MonoLLVMDir)' == ''">mono.llvm+</DefaultMonoSubsets>
     <DefaultMonoSubsets Condition="'$(MonoAOTEnableLLVM)' == 'true' and '$(MonoAOTLLVMDir)' == ''">mono.llvm+</DefaultMonoSubsets>
     <DefaultMonoSubsets Condition="'$(TargetOS)' == 'Browser'">$(DefaultMonoSubsets)mono.wasmruntime+</DefaultMonoSubsets>
-    <DefaultMonoSubsets Condition="'$(BuildMonoCrossAOT)' == 'true'">$(DefaultMonoSubsets)mono.aotcross+</DefaultMonoSubsets>
+    <DefaultMonoSubsets Condition="'$(BuildMonoCrossAOT)' != ''">$(DefaultMonoSubsets)mono.aotcross+</DefaultMonoSubsets>
```
Comment (ViktorHofer) [inline eng/Subsets.props:48]: Curious, why don't you check on true not true anymore?

Comment (directhex) [reply eng/Subsets.props:48]: It's a string not a bool now. I don't want to test `"+android+browser+" == "true"`, I want to test "Is this not null?"

Comment (akoeplinger) [reply eng/Subsets.props:48]: maybe we should rename it to `MonoCrossAOTBuildTargets` to make that clearer?

Comment (akoeplinger) [reply eng/Subsets.props:48]: or `MonoCrossAOTTargetOS` like we use in the yaml

File: eng/pipelines/mono/templates/generate-offsets.yml [hunk 2]
```diff
@@ -62,10 +62,10 @@ jobs:
 
     # Build
     - ${{ if ne(parameters.osGroup, 'windows') }}:
-      - script: ./build$(scriptExt) -subset mono.aotcross -c $(buildConfig) -arch $(archType) $(osOverride) -ci $(officialBuildIdArg) /p:MonoGenerateOffsetsOnly=true
+      - script: ./build$(scriptExt) -subset mono.aotcross -c $(buildConfig) -arch $(archType) $(osOverride) -ci $(officialBuildIdArg) /p:MonoGenerateOffsetsOnly=$(osGroup)
```
Comment (akoeplinger) [inline eng/pipelines/mono/templates/generate-offsets.yml:65]: same here, I think something like `MonoGenerateOffsetsOSGroups` makes more sense

**Review by akoeplinger — APPROVED** [review]
Looks good apart from the naming comments

**Comment by directhex** [comment]
Merging despite the failure, as the failure is unrelated infra https://github.com/dotnet/runtime/issues/47185

**Pull request merged**